lib/backend/lite/lite.go

```diff
@@ -629,6 +630,95 @@ func (l *Backend) getInTransaction(ctx context.Context, key backend.Key, tx *sql
 	return nil
 }
 
+func (l *Backend) Items(ctx context.Context, params backend.IterateParams) iter.Seq2[backend.Item, error] {
+	if params.StartKey.IsZero() {
+		err := trace.BadParameter("missing parameter startKey")
+		return func(yield func(backend.Item, error) bool) { yield(backend.Item{}, err) }
+	}
+	if params.EndKey.IsZero() {
+		err := trace.BadParameter("missing parameter endKey")
+		return func(yield func(backend.Item, error) bool) { yield(backend.Item{}, err) }
+	}
+
+	const (
+		queryAsc        = "SELECT key, value, expires, revision FROM kv WHERE (key >= ? and key <= ? and key > ?) AND (expires is NULL or expires > ?) ORDER BY key ASC LIMIT ?"
+		queryDesc       = "SELECT key, value, expires, revision FROM kv WHERE (key >= ? and key <= ? and key < ?) AND (expires is NULL or expires > ?) ORDER BY key DESC LIMIT ?"
+		defaultPageSize = 1000
+	)
+	return func(yield func(backend.Item, error) bool) {
+		limit := params.Limit
+		if limit <= 0 {
+			limit = backend.DefaultRangeLimit
+		}
+
+		var exclusiveStartKey string
+		startKey := params.StartKey.String()
+		endKey := params.EndKey.String()
+
+		query := queryAsc
+		if params.Descending {
+			exclusiveStartKey = endKey
+			query = queryDesc
+		}
+
+		var pageLimit, pageCount, totalCount int
+		items := make([]backend.Item, defaultPageSize)
+		for {
+			pageLimit = min(limit-totalCount, defaultPageSize)
```

Comment:
The language can enforce lengths more reliably than us.

```suggestion
		var totalCount int
		items := make([]backend.Item, 0, defaultPageSize)
		for {
			items = items[:0]
			pageLimit := min(limit-totalCount, defaultPageSize)
```